libshortfin/python/array_host_ops.cc

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 Advanced Micro Devices, Inc.
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "./lib_ext.h"
+#include "./utils.h"
+#include "shortfin/array/api.h"
+#include "shortfin/support/logging.h"
+#include "xtensor/xsort.hpp"
+#include "xtl/xhalf_float.hpp"
+
+using namespace shortfin::array;
+
+namespace shortfin::python {
+
+namespace {
+
+static const char DOCSTRING_ARGMAX[] =
+    R"(Returns the indices of the maximum values along an axis.
+
+See documentation for numpy.argmax which this operator emulates.
+
+Implemented dtypes: float16, float32.
+)";
```

Comment:
Can we add a note somewhere that out must be / will be int64? It's not obvious here. Also, would be good to document what divise_visible means.

Also, maybe differing opinion, personally, I prefer the docs to not delegate semantics to something else. I don't think argmax docs are going to change much, can we just copy them here from numpy and mention the semantics mimic numpy?

Also, does nanobind allow specifying parameter specific docs? I like that numpy docs has a parameter section for each operator and has per parameter docs.